File: src/main/java/me/miki/shindo/management/notification/NotificationManager.java
package me.miki.shindo.management.notification;

import me.miki.shindo.Shindo;
import me.miki.shindo.management.language.TranslateText;

import java.util.concurrent.LinkedBlockingQueue;

public class NotificationManager {

    private final LinkedBlockingQueue<Notification> notifications = new LinkedBlockingQueue<Notification>();

    public NotificationManager() {
        Shindo.getInstance().getEventManager().register(new NotificationHandler(notifications));
    }

    public void post(TranslateText title, TranslateText message, NotificationType type) {
        notifications.add(new Notification(title, message, type));
    }

    public void post(String title, String message, NotificationType type) {
        notifications.add(new Notification(title, message, type));
    }

    public void post(TranslateText title, String message, NotificationType type) {
        notifications.add(new Notification(title, message, type));
    }
}

File: src/main/kotlin/me/miki/shindo/management/notification/NotificationManager.kt
package me.miki.shindo.management.notification

import me.miki.shindo.Shindo
import me.miki.shindo.logger.FileLogWriter
import me.miki.shindo.management.language.TranslateText
import java.util.concurrent.LinkedBlockingQueue

class NotificationManager {
    private val notifications = LinkedBlockingQueue<Notification>()

    init {
        Shindo.getInstance().getEventManager().register(NotificationHandler(notifications))
    }

    fun post(
        title: TranslateText,
        message: TranslateText,
        type: NotificationType,
    ) {
        notifications.add(Notification(title, message, type))
        FileLogWriter.notification(title.getText() + " | " + message.getText() + " | " + type.name)
    }

    fun post(
        title: String,
        message: String,
        type: NotificationType,
    ) {
        notifications.add(Notification(title, message, type))
        FileLogWriter.notification(title + " | " + message + " | " + type.name)
    }

    fun post(
        title: TranslateText,
        message: String,
        type: NotificationType,
    ) {
        notifications.add(Notification(title, message, type))
        FileLogWriter.notification(title.getText() + " | " + message + " | " + type.name)
    }

    fun post(
        title: String,
        message: TranslateText,
        type: NotificationType,
    ) {
        notifications.add(Notification(title, message, type))
        FileLogWriter.notification(title + " | " + message.getText() + " | " + type.name)
    }
}
